pkg/config/apiserver.go

```diff
@@ -0,0 +1,44 @@
+package config
+
+import (
+	"net/url"
+	"strconv"
+)
+
+type ApiServer struct {
+	// SubjectAltNames added to API server certs
+	SubjectAltNames []string `json:"subjectAltNames"`
+	// Kube apiserver advertise address to work around the certificates issue
+	// when requiring external access using the node IP. This will turn into
+	// the IP configured in the endpoint slice for kubernetes service. Must be
+	// a reachable IP from pods. Defaults to service network CIDR first
+	// address.
+	AdvertiseAddress string `json:"advertiseAddress,omitempty"`
+	// Determines if kube-apiserver controller should configure the
+	// AdvertiseAddress in the loopback interface. Automatically computed.
+	SkipInterface bool `json:"-"`
+
+	// The URL of the API server
+	URL string `json:"-"`
+}
+
+// extract the api server port from the cluster URL
+func (a *ApiServer) Port() (int, error) {
+	var port string
+
+	parsed, err := url.Parse(a.URL)
```

Comment:
If the `ApiServer.URL` and `Port` value is an internally created one and not changing at runtime, what do you think about changing the type to `URL` and populate a `Port` attribute on instantiation, that way we push out the error check so it's not on the caller to handle?

```go
type ApiServer struct {
        ...
	// The URL of the API server
	URL  *url.URL `json:"-"`
        // The Port of the API server
        Port int      `json:"-"`
}
```

Reply:
I think I like that. I can experiment with it when I rebase this on #1506

Reply:
I decided to stick with a string for the URL, since that's how it's used, and added a separate field for the port.